manager container goes in panic with SIGSEGV

Once I create the `RouteMigrate` object, the route is migrated from a namespace to another, but the `manager` container of `dioscuri-controller` goes in error and in crashloopback status because SIGSEGV.
Here the logs

`20-03-26T17:25:32.388Z	INFO	controllers.RouteMigrate	Beginning route migration checks for routes in blue-env moving to green-env	{"routemigrate": "blue-env/active-standby"}
2020-03-26T17:25:33.132Z	INFO	controllers.RouteMigrate	Attempting to migrate route production-www-route - bluegreen.192.168.42.216.nip.io	{"routemigrate": "blue-env"}
2020-03-26T17:25:33.132Z	INFO	controllers.RouteMigrate	Getting existing route production-www-route in namespace green-env	{"routemigrate": "blue-env"}
2020-03-26T17:25:33.132Z	INFO	controllers.RouteMigrate	Creating route production-www-route in namespace green-env	{"routemigrate": "blue-env"}
2020-03-26T17:25:33.139Z	INFO	controllers.RouteMigrate	Removing old route production-www-route in namespace blue-env	{"routemigrate": "blue-env"}
2020-03-26T17:25:33.149Z	INFO	controllers.RouteMigrate	Done migrating route production-www-route	{"routemigrate": "blue-env"}
E0326 17:25:33.149789       1 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 236 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic(0x13b76a0, 0x215cd10)
	/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190913080033-27d36303b655/pkg/util/runtime/runtime.go:74 +0xa3
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190913080033-27d36303b655/pkg/util/runtime/runtime.go:48 +0x82
panic(0x13b76a0, 0x215cd10)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/amazeeio/dioscuri/controllers.(*RouteMigrateReconciler).Reconcile(0xc000529050, 0xc000602160, 0x8, 0xc000602140, 0xe, 0xc0000ddcd8, 0xc000428120, 0xc000428098, 0x1727800)
	/workspace/controllers/routemigrate_controller.go:176 +0x17d5
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc00036a180, 0x1402580, 0xc0005caf20, 0xc00020e500)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/internal/controller/controller.go:256 +0x162
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc00036a180, 0x0)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/internal/controller/controller.go:232 +0xcb
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker(0xc00036a180)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/internal/controller/controller.go:211 +0x2b
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc000360100)
	/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190913080033-27d36303b655/pkg/util/wait/wait.go:152 +0x5e
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000360100, 0x3b9aca00, 0x0, 0x1, 0xc000046180)
	/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190913080033-27d36303b655/pkg/util/wait/wait.go:153 +0xf8
k8s.io/apimachinery/pkg/util/wait.Until(0xc000360100, 0x3b9aca00, 0xc000046180)
	/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190913080033-27d36303b655/pkg/util/wait/wait.go:88 +0x4d
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/internal/controller/controller.go:193 +0x328
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference`

`[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x120fe55]
goroutine 236 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190913080033-27d36303b655/pkg/util/runtime/runtime.go:55 +0x105
panic(0x13b76a0, 0x215cd10)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/amazeeio/dioscuri/controllers.(*RouteMigrateReconciler).Reconcile(0xc000529050, 0xc000602160, 0x8, 0xc000602140, 0xe, 0xc0000ddcd8, 0xc000428120, 0xc000428098, 0x1727800)
	/workspace/controllers/routemigrate_controller.go:176 +0x17d5
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc00036a180, 0x1402580, 0xc0005caf20, 0xc00020e500)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/internal/controller/controller.go:256 +0x162
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc00036a180, 0x0)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/internal/controller/controller.go:232 +0xcb
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker(0xc00036a180)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/internal/controller/controller.go:211 +0x2b
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc000360100)
	/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190913080033-27d36303b655/pkg/util/wait/wait.go:152 +0x5e
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000360100, 0x3b9aca00, 0x0, 0x1, 0xc000046180)
	/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190913080033-27d36303b655/pkg/util/wait/wait.go:153 +0xf8
k8s.io/apimachinery/pkg/util/wait.Until(0xc000360100, 0x3b9aca00, 0xc000046180)
	/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190913080033-27d36303b655/pkg/util/wait/wait.go:88 +0x4d
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/internal/controller/controller.go:193 +0x328
`   

I'm testing it by building the image from `master` branch and with `minishift v1.33.0+ba29431`

--- a/controllers/routemigrate_controller.go
+++ b/controllers/routemigrate_controller.go
@@ -173,8 +173,10 @@ func (r *RouteMigrateReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error
 				}
 				migratedRoutes = append(migratedRoutes, MigratedRoutes{NewRoute: newRoute, OldRouteNamespace: sourceNamespace})
 				routeScheme := "http://"
-				if route.Spec.TLS.Termination != "" {
-					routeScheme = "https://"
+				if route.Spec.TLS != nil {
+					if route.Spec.TLS.Termination != "" {
+						routeScheme = "https://"
+					}
 				}
 				activeMigratedRoutes = append(activeMigratedRoutes, fmt.Sprintf("%s%s", routeScheme, route.Spec.Host))
 			}
@@ -192,8 +194,10 @@ func (r *RouteMigrateReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error
 				// add the migrated route so we go through and fix them up later
 				migratedRoutes = append(migratedRoutes, MigratedRoutes{NewRoute: newRoute, OldRouteNamespace: destinationNamespace})
 				routeScheme := "http://"
-				if route.Spec.TLS.Termination != "" {
-					routeScheme = "https://"
+				if route.Spec.TLS != nil {
+					if route.Spec.TLS.Termination != "" {
+						routeScheme = "https://"
+					}
 				}
 				standbyMigratedRoutes = append(standbyMigratedRoutes, fmt.Sprintf("%s%s", routeScheme, route.Spec.Host))
 			}

